packages/react-native-executorch/common/rnexecutorch/models/speech_to_text/MoonshineStrategy.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "SpeechToTextStrategy.h"
+
+namespace rnexecutorch {
+
+class MoonshineStrategy : public SpeechToTextStrategy {
+public:
+  std::pair<TensorPtr, std::vector<int32_t>>
+  prepareAudioInput(std::span<float> waveform) override;
+
+  TensorPtr prepareTokenInput(const std::vector<int64_t> &prevTokens) override;
+
+  std::string getDecoderMethod() const override { return "forward_cached"; }
+
+  int64_t extractOutputToken(const void *outputPtr,
+                             std::vector<int32_t> &sizes) override;
+};
```

Comment:
Make sure, that you marked everything you can with `const`, `noexcept`, and `static` qualifiers. Also you don't have constructor cleared here, so maybe consider removing all of them explicitly.